security/s2a/internal/crypter/aesgcm_test.go

```diff
@@ -80,176 +38,142 @@ func getGCMCryptoPair(key []byte, t *testing.T) (S2AAeadCrypter, S2AAeadCrypter)
 	return client, server
 }
 
-func parseWycheProofTestVectors(jsonFilePath string, t *testing.T) []cryptoTestVector {
-	jsonFile, err := os.Open(jsonFilePath)
-	if err != nil {
-		t.Fatalf("failed to open wycheproof json test vectors file: %v", err)
-	}
-	defer jsonFile.Close()
-
-	dec := json.NewDecoder(jsonFile)
-
-	var tf testFile
-	err = dec.Decode(&tf)
-	if err != nil {
-		t.Fatalf("failed to decode wycheproof json file: %v", err)
-	}
-
-	var testVectors []cryptoTestVector
-	for _, testGroup := range tf.TestGroups {
-		// Skip over unsupported inputs.
-		if isUnsupportedInput(testGroup.IvSize, testGroup.KeySize, testGroup.TagSize) {
-			continue
-		}
-		for _, test := range testGroup.Tests {
-			testVectors = append(testVectors, cryptoTestVector{
-				key:         dehex(test.Key),
-				plaintext:   dehex(test.Msg),
-				ciphertext:  dehex(test.Ct),
-				tag:         dehex(test.Tag),
-				nonce:       dehex(test.Iv),
-				aad:         dehex(test.Aad),
-				result:      test.Result,
-				comment:     test.Comment,
-				id:          test.TcId,
-				allocateDst: true,
-			})
-		}
-	}
-
-	return testVectors
-}
-
 func isFailure(result string, err error, got, expected []byte) bool {
-	return (result == validResult && (err != nil || !bytes.Equal(got, expected))) ||
-		(result == invalidResult && bytes.Equal(got, expected))
+	return (result == ValidResult && (err != nil || !bytes.Equal(got, expected))) ||
+		(result == InvalidResult && bytes.Equal(got, expected))
 }
 
-func isUnsupportedInput(ivSize, keySize, tagSize int) bool {
-	return ivSize != 96 || (keySize != 128 && keySize != 256) || tagSize != 128
+func wycheProofTestVectorFilter(testGroup TestGroup) bool {
+	return testGroup.IvSize != 96 ||
+		(testGroup.KeySize != 128 && testGroup.KeySize != 256) ||
+		testGroup.TagSize != 128
 }
 
 func TestWycheProofTestVectors(t *testing.T) {
```

Comment:
Should this function start with a lowercase `t` rather than `T`? I don't think we want this function to be used outside of this file, correct?

Reply:
Go unit tests should start with the keyword `Test`(with a capital T) for the testing framework to recognize it as a unit test.

Reply:
Good point. We shall leave it as is then :)